packages/libraries/client/tests/usage-collector.spec.ts

```diff
@@ -336,6 +336,31 @@ test('(processVariables: true) collect used-only input fields', async () => {
   expect(info.fields).not.toContain(`PaginationInput.offset`);
 });
 
+test('(processVariables: true) should not collect input when corresponding variable is not provided', async () => {
+  const collect = createCollector({
+    schema,
+    max: 1,
+    processVariables: true,
+  });
+  const info = collect(
+    parse(/* GraphQL */ `
+      query getProjects($pagination: PaginationInput, $type: ProjectType!) {
+        projects(filter: { pagination: $pagination, type: $type }) {
+          id
+        }
+      }
+    `),
+    {
+      type: 'STITCHING',
+    }
+  ).value;
+
+  expect(info.fields).toContain(`FilterInput.type`);
+  expect(info.fields).toContain(`FilterInput.pagination`);
+  expect(info.fields).not.toContain(`PaginationInput.limit`);
+  expect(info.fields).not.toContain(`PaginationInput.offset`);
+});
```

Comment:
hmmm, if it's not collecting `PaginationInput` then when `PaginationInput` is removed (because it's safe from the usage reports perspective) this query will fail, no? Because it contains `PaginationInput` and it's not defined.

Reply:
oh, yes, you're right. I need to think about it and find a way to handle this case